### PR TITLE
Bst

### PR DESCRIPTION
**Pull request by constanthatz**

### CODE REVIEW

File: bst.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+if __name__ == '__main__':
+    import timeit
+
+    nums = []
+    nums.append(range(0, 5))
+    nums.append(range(0, 50))
+    nums.append(range(0, 500))
+
+    for item in nums:
+        num = item
```
Comment (cewing) [inline bst.py:122]: why this instead of `for num in nums`?

File: test_bst.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    assert test_tree_large.depth() == 5
+
+
+def test_balance(test_tree):
+    assert test_tree.balance() == 0
+
+
+def test_balance_negative(test_tree_large):
+    assert test_tree_large.balance() < 0
+
```
Comment (cewing) [inline test_bst.py:92]: You might wish to add a test that asserts that the balance is the expected value, rather than just more or less than zero.  For balancing the tree, being sure you have the right value for balance will become important.
